source/s3_auto_ranged_get.c

```diff
@@ -400,9 +448,20 @@ static void s_s3_auto_ranged_get_request_finished(
             (auto_ranged_get->synced_data.num_parts_successful + auto_ranged_get->synced_data.num_parts_failed),
             auto_ranged_get->synced_data.total_num_parts);
     } else {
-        ++auto_ranged_get->synced_data.num_parts_failed;
-
-        aws_s3_meta_request_set_fail_synced(meta_request, request, error_code);
+        if (s_check_empty_file_download_error_synced(request)) {
+            AWS_LOGF_DEBUG(
+                AWS_LS_S3_META_REQUEST,
+                "id=%p Getting an empty file, create a new request without range header to fetch the empty "
+                "file",
+                (void *)meta_request);
+            auto_ranged_get->synced_data.get_without_range = true;
+            ++meta_request->synced_data.num_parts_delivery_completed;
```

Comment:
maybe I should update the num parts failed and invoke the `s_s3_meta_request_delivered_requests` instead?

Reply:
I think we'll need to put the if/else that starts on line 434 inside of an if(request_tag == AWS_S3_AUTO_RANGE_GET_REQUEST_TYPE_PART).  And then we can have a separate block for if(request_tag == AWS_S3_AUTO_RANGE_GET_REQUEST_TYPE_PART_WITHOUT_RANGE) which sets request_without_range_completed to true, and then calls aws_s3_meta_request_set_fail_synced if the request completed with an error.